src/DependencyInjection/LazyContainerFactory.php

```diff
@@ -729,9 +729,12 @@ static function (
             }
 
             $skippedClassResolver = new SkippedClassResolver();
-            $skippedClasses = array_keys($skippedClassResolver->resolve());
-            foreach ($skippedClasses as $skippedClass) {
-                ContainerMemento::forgetService($container, $skippedClass);
+            $skippedElements = $skippedClassResolver->resolve();
+            foreach ($skippedElements as $skippedClass => $path) {
+                // completely forget the Rector rule only when no path specified
+                if ($path === null) {
+                    ContainerMemento::forgetService($container, $skippedClass);
```

Comment:
could you give me an idea why we need to drop classes from the container when classes are skipped?

Reply:
This @TomasVotruba blog post may be the reason https://tomasvotruba.com/blog/removing-service-from-laravel-container-is-not-that-easy